config/compression.go

```diff
@@ -0,0 +1,84 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Compression struct {
+	Request  ReqCompression  `mapstructure:"request"`
+	Response RespCompression `mapstructure:"response"`
+}
+
+// CompressionConfig defines if compression is enabled and what type of compression to use
+type ReqCompression struct {
+	Enabled bool     `mapstructure:"enabled"`
+	Kind    []string `mapstructure:"kind,flow"`
+	kindMap map[CompressionKind]bool
```

Comment:
`Validate` function on line 55 checks if kind is valid. if yes then adds value in `kindMap`. 

Seeing this usage,  `kindMap map[CompressionKind]bool` can be changed to have `map[CompressionKind] struct{}`. Usage of `struct{}` benefits with memory optimisation 

```
func (cfg *ReqCompression) Validate(errs []error) []error {
...
  for _, kind := range cfg.Kind {
	k := CompressionKind(strings.ToLower(kind))
	if !k.IsValid() {
		errs = append(errs, fmt.Errorf("compression type %s is not valid", kind))
	} else {
		- cfg.kindMap[k] = true
		+ cfg.kindMap[k] = struct{}{}
	}
 }
..
}
```

Reply:
Oh yes, true. I almost forgot about this little trick. Thanks!